reflection/src/main/java/com/alevel_hw2/ContactDAO.java

```diff
@@ -0,0 +1,117 @@
+package com.alevel_hw2;
+
+import java.sql.*;
+
+public class ContactDAO {
+    private final String CREATE_TABLE_QUERY = "CREATE TABLE IF NOT EXISTS contacts (" +
+            "id bigserial PRIMARY KEY," +
+            "contact VARCHAR(255)," +
+            "contact_type VARCHAR(255))";
+
+    private final String CREATE_CONTACT = "INSERT INTO contacts (contact, contact_type) VALUES (?, ?)";
+
+    private final String UPDATE_CONTACT = "UPDATE contacts SET contact = ?, contact_type = ? WHERE id = ?;";
+
+    private final String DELETE_CONTACT = "DELETE FROM contacts WHERE id = ?";
+
+    private final String READ_CONTACT = "SELECT * FROM contacts WHERE id = ?";
+
+    private final String READ_ALL_CONTACT = "SELECT * FROM contacts";
+
+    private Connection getConnection() {
+        try {
+            Connection connection = DriverManager.getConnection("jdbc:postgresql://localhost:5432/alevel",
+                    "postgres", "postgres");
+            return connection;
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public void createTable() {
+        try (Connection connection = getConnection()) {
+            PreparedStatement preparedStatement = connection.prepareStatement(CREATE_TABLE_QUERY);
+            preparedStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public void createContact(String contact, String contact_type) {
+        try (Connection connection = getConnection()) {
+            PreparedStatement preparedStatement = connection.prepareStatement(CREATE_CONTACT);
+
+            preparedStatement.setString(1, contact);
+            preparedStatement.setString(2, contact_type);
+
+            preparedStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public void readContact(long id) {
+        try (Connection connection = getConnection()) {
+            PreparedStatement preparedStatement = connection.prepareStatement(READ_CONTACT);
+            preparedStatement.setLong(1, id);
+            try (ResultSet resultSet = preparedStatement.executeQuery()) {
+                if (resultSet.next()) {
+                    int idContact = resultSet.getInt("id");
+                    String contact = resultSet.getString("contact");
+                    String contact_type = resultSet.getString("contact_type");
+                    System.out.println("Contact ID: " + idContact + ", Contact: " + contact + ", Contact_type: "
+                            + contact_type);
+                }
```

Comment:
here the same as well. Map this to Contact object and return it.

And, you may notice, that mapping in 2 methods are the same, so you can move it to separate private method that accept ResultSet and build object.